### PR TITLE
newt: Fix for Linuxbrew

### PR DESCRIPTION
**Pull request by ProfFan**
- [x] Have you followed the [guidelines for contributing](https://github.com/Linuxbrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Linuxbrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

This patch should fix the Linuxbrew-installed newt's Python bindings.

### CODE REVIEW

**Comment by maxim-belkin** [comment]
what is exactly the problem
Also, for a single change use
```
inreplace ... unless OS.mac?
```

**Comment by maxim-belkin** [comment]
I'd ask @sjackman for a review

**Comment by ProfFan** [comment]
@maxim-belkin The patch is to fix the proper building of `newt`'s python bindings. The `newt` library's python module `snack` is used by `byobu-config` and a lot of other CLI programs to display a GUI interface.

**Comment by rwhogg** [comment]
> The patch is to fix the proper building of newt's python bindings. The newt library's python module snack is used by byobu-config and a lot of other CLI programs to display a GUI interface.

Indeed, see https://github.com/Linuxbrew/homebrew-core/pull/2140.

File: Formula/newt.rb [hunk 1]
```diff
@@ -31,6 +31,10 @@ def install
       s.gsub! "`$$pyconfig --libs`", '""'
     end if OS.mac?
 
+    if OS.linux?
+      inreplace "configure", "/usr/include/python", "#{HOMEBREW_PREFIX}/include/python"
+    end
```
Comment (sjackman) [inline Formula/newt.rb]: Does it work if you remove this chunk, replace it with a dependency, and use `--env=super`?
```ruby
depends_on :python unless OS.mac?
```
```sh
brew install --env=super newt
```

**Comment by sjackman** [comment]
`newt` built correctly and produced a bottle but `brew test newt` gave this error:
CircleCI
```
==> brew test newt --verbose
Testing newt
==> python -c import snack
.
Last 150 lines from /home/ubuntu/.cache/Homebrew/Logs/newt/test.01.python:
2017-04-03 14:51:13 +0000

python
-c
import snack

Traceback (most recent call last):
  File "<string>", line 1, in <module>
ImportError: No module named snack
Error: newt: failed
Failed executing: python -c import\ snack
```
https://circleci.com/gh/Linuxbrew/homebrew-core/4738

Note that using `--env=super` does not affect `brew test`, which uses `--env=std`.

File: Formula/newt.rb (hunk 1 above)
Comment (ProfFan) [inline Formula/newt.rb]: I think it will not, cause this value is hardcoded in the autoconf `configure.ac` script.

**Comment by ProfFan** [comment]
@sjackman I got the reason. The PR only works when you build `newt` in source as the path depends on your `$HOME`. I will fix this ASAP.

**Comment by ProfFan** [comment]
@sjackman Fixed.

File: Formula/newt.rb [hunk 2]
```diff
@@ -14,9 +14,10 @@ class Newt < Formula
   depends_on "gettext"
   depends_on "popt"
   depends_on "s-lang"
+  depends_on :python unless OS.mac?
 
   def install
-    args = ["--prefix=#{prefix}", "--without-tcl"]
+    args = ["--prefix=#{prefix}", "--without-tcl", "--with-python"]
```
Comment (sjackman) [inline Formula/newt.rb]: The existing behaviour should be maintained on macOS.
```ruby
args = ["--prefix=#{prefix}", "--without-tcl"]
args << "--with-python" unless OS.mac?
```
Do you know why this change is necessary on Linux but not on macOS?

**Comment by sjackman** [comment]
Glad to see it's building! One more change, then it's good to go.

File: Formula/newt.rb (hunk 2 above)
Comment (ProfFan) [inline Formula/newt.rb]: It seems that this change is unnecessary. Will fix asap.

File: Formula/newt.rb [hunk 3]
```diff
@@ -31,6 +32,8 @@ def install
       s.gsub! "`$$pyconfig --libs`", '""'
     end if OS.mac?
 
+    inreplace "configure", "/usr/include/python", "#{HOMEBREW_PREFIX}/include/python" unless OS.mac?
```
Comment (sjackman) [inline Formula/newt.rb]: There's no directory on my system named `"#{HOMEBREW_PREFIX}/include/python`. Should it be instead
```ruby
"#{Formula["python"].include}/"python2.7"}"
```
?

Comment (ProfFan) [inline Formula/newt.rb]: Please look inside the `configure` script. The original statement is `/usr/include/python*`, so it can find all installed python versions.

**Comment by sjackman** [comment]
Merged. Thanks for your contribution to Linuxbrew, Fan!
